src/ChangeStream.php

```diff
@@ -234,6 +242,13 @@ private function resume()
         $newChangeStream = call_user_func($this->resumeCallable, $this->resumeToken);
         $this->csIt = $newChangeStream->csIt;
         $this->csIt->rewind();
-        $this->onIteration(false);
+        /* Note: if we are resuming after a call to rewind(), $hasAdvanced will
+         * always be false. If it were true, then the previous call to
+         * IteratorIterator::rewind() would either have been a NOP and succeeded
+         * because the cursor was still at its first position or have thrown a
+         * LogicException for attempting to rewind an advanced cursor. In the
+         * latter case, the exception would have bypassed the catch block for
+         * resuming. */
+        $this->onIteration($this->hasAdvanced);
```

Comment:
I'd appreciate a sanity check on this. While coming up with the current implementation in this PR, I ran the test suite with a hard assertion of `$this->hasAdvanced === false` placed before the call to `resume()` in `rewind()`'s `catch` block, and it was never triggered.

Reply:
I'm not sure I follow this part:
```
If it were true, then the previous call to IteratorIterator::rewind() would either have been
a NOP and succeeded because the cursor was still at its first position
```
How can `$hasAdvanced` be true if the cursor was still at its first position?

Reply:
My point here was that "if it were true" cannot happen in practice. Either the `rewind()` succeeds as a NOP and doesn't throw, so we never call `resume()` in the first place, or it throws a LogicException, which we don't catch (and also don't call `resume()`). I'll reword this a bit to better explain that, as I feel I only conveyed that the LogicException case is when we never reach this point.

> How can `$hasAdvanced` be true if the cursor was still at its first position?

If this was a more general question (apart from `resume()` being invoked), I can address that. When a cursor is first created (basic or tailable alike), it starts before the first element. `valid()` is `false` and there is no current element. The initial call to `rewind()` or `next()` is responsible for advancing to the first position (if any), updating `valid()`, and optionally populating the current element.

We use `$hasAdvanced` here to track if we've advanced before the initial pre-iteration state. If it's more helpful, I suppose we can rename this to `$hasIterationStarted`.

Another edge case is that it's kosher to call `rewind()` multiple times in sequence. And since `$key` is initialized to `0`, we should never increment the key after a successful call to `rewind()`, since, unlike `next()` or resuming, `rewind()` itself never moves past the first element.

Reply:
Oh I see, thanks for the explanation! All of this makes sense.

Reply:
> I suppose we can rename this to `$hasIterationStarted`.

Ah, I forgot that I actually added a doc block to `$hasAdvanced` to explain exactly what it implies ([here](https://github.com/mongodb/mongo-php-library/pull/626/files#diff-d407b16b9acca873bbbaf0198a74fe57R54)). I think that's sufficient, as `$hasIterated` might also sound confusing since calls to `rewind()` and `next()` can still leave it `false` if the change stream doesn't come across a change event. I'll stick with the shorter name and just revise this comment block to be more clear.

Reply:
Note: this is a slight edge case where `rewind()` _could_ be called immediately after resuming without encountering a LogicException from the driver. Since resuming creates a fresh cursor and rewinds it, a user could technically call `ChangeStream::rewind()` after a resume and not see an error. This is very trivial and may not be worth fixing, but I opened [PHPLIB-448](https://jira.mongodb.org/browse/PHPLIB-448) to track it.